### PR TITLE
add upernet algo

### PR DESCRIPTION
**Pull request by haiasd**

### CODE REVIEW

File: easycv/models/segmentation/encoder_decoder.py [hunk 1]
```diff
@@ -55,12 +56,32 @@ def __init__(
...
+
+    def init_weights(self):
+        logger = get_root_logger()
+        if isinstance(self.pretrained, str):
+            load_checkpoint(
+                self.backbone, self.pretrained, strict=False, logger=logger)
+        elif self.pretrained:
+            if self.backbone.__class__.__name__ == 'PytorchImageModelWrapper':
+                self.backbone.init_weights(pretrained=self.pretrained)
+            elif hasattr(self.backbone, 'default_pretrained_model_path'
```
Comment (Cathy0908) [inline easycv/models/segmentation/encoder_decoder.py:69]: When is `default_pretrained_model_path` required?

File: easycv/models/segmentation/heads/uper_head.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            upsampled_ppm_out = resize(
+                ppm_out,
+                size=x.size()[2:],
+                mode='bilinear',
+                align_corners=self.align_corners)
+            ppm_outs.append(upsampled_ppm_out)
+        return ppm_outs
+
+
+def resize(input,
```
Comment (Cathy0908) [inline easycv/models/segmentation/heads/uper_head.py:196]: The resize api has been implemented in https://github.com/alibaba/EasyCV/blob/master/easycv/models/utils/ops.py?plain=1#L9

File: easycv/models/segmentation/encoder_decoder.py (hunk 1 above)
Comment (haiasd) [reply easycv/models/segmentation/encoder_decoder.py:69]: https://github.com/alibaba/EasyCV/blob/master/easycv/models/modelzoo.py

File: easycv/models/segmentation/heads/uper_head.py (hunk 2 above)
Comment (haiasd) [reply easycv/models/segmentation/heads/uper_head.py]: done

Comment (Cathy0908) [reply easycv/models/segmentation/heads/uper_head.py]: LGTM

File: docs/source/model_zoo_seg.md [hunk 3]
```diff
@@ -7,3 +7,10 @@ Pretrained on **Pascal VOC 2012 + Aug**.
...
+
+## UperNet
+
+Pretrained on **Pascal VOC 2012 + Aug**.
+| Algorithm  | Config                                                       | mIoU | Download                                                     |
+| ---------- | ------------------------------------------------------------ | ------------------------ | ------------------------------------------------------------ |
+| upernet_r50 | [upernet_r50_512x512_8xb4_60e_voc12aug](https://github.com/alibaba/EasyCV/tree/master/configs/segmentation/upernet/upernet_r50_512x512_8xb4_60e_voc12aug.py) | 76.59               | [model](http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/segmentation/upernet_r50/epoch_60.pth) - [log](http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/segmentation/upernet_r50/20220706_114712.log.json) |
```
Comment (wenmengzhou) [inline docs/source/model_zoo_seg.md:16]: 只有这一个type么， 可以多跑一些

File: tests/models/segmentation/heads/test_uper_head.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Alibaba, Inc. and its affiliates.
+import unittest
+
+import torch
+
+from easycv.models.builder import build_head
+
+
+class FCNHeadTest(unittest.TestCase):
```
Comment (wenmengzhou) [inline tests/models/segmentation/heads/test_uper_head.py:9]: change test name

**Review by wenmengzhou — COMMENTED** [review]
add a notebook for training and inference and visualization

File: docs/source/model_zoo_seg.md (hunk 3 above)
Comment (haiasd) [reply docs/source/model_zoo_seg.md:16]: I may add benchmark in another CR, after double check parameter setting

File: tests/models/segmentation/heads/test_uper_head.py (hunk 4 above)
Comment (haiasd) [reply tests/models/segmentation/heads/test_uper_head.py]: done

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**